clouddriver-elasticsearch/src/main/java/com/netflix/spinnaker/clouddriver/elasticsearch/ops/BulkUpsertEntityTagsAtomicOperation.java

```diff
@@ -243,9 +243,15 @@ private static void replaceTagContents(EntityTags currentTags, EntityTags entity
     }
   }
 
-  private static EntityTags.EntityTagMetadata tagMetadata(String tagName, Date now) {
+  private static EntityTags.EntityTagMetadata tagMetadata(EntityTags.EntityTag entityTag, Date now) {
     String user = AuthenticatedRequest.getSpinnakerUser().orElse("unknown");
 
+    String tagName = entityTag.getName();
+    if (entityTag.getTimestamp() != null) {
+      // entity tag has an explicit timestamp, favor it for last modified date
+      now = new Date(entityTag.getTimestamp());
```

Comment:
if we're creating an entity tag for some external event, this lets us use the timestamp of the external event rather than `now`.